### DISCUSSION TITLE
PS/2 2.88MB Enhanced Interface and Media Sense?

### DISCUSSION THREAD

**Discussion by WBSTClarke** [keirf/flashfloppy#553] [Q&A]
Hi Keir,

first of all, many thanks for your detailed and impressive work on this. It has been extremely helpful to our PS/2 Enthusiast community. Most of these are 25-35 years old and electrolytic capacitor failures are common in the ir floppy drives.

I've been wondering if the interface pins of the PS/2 2.88MB Media Sense Drive and Secure Drive (nee Electronic Eject), as per the 34-pin Enhanced Interface at the end of:
https://www.ardent-tool.com/docs/pdf/ps2_fdd_trm_s42g2194_00.pdf
are even connected/controllable via GPIO pins, and on which Gotek PCB designs and with which related MCUs?

In particular, I'm referring to the non-secure 2.88MB drive identification - Drive Type(1,0) - and concomitant resultant Media Sense(1,0) indications, based upon the selected image size (2880KB, 1440KB or 720KB) and/or image type qualifier (*.ed.*.img, *.hd.*.img etc.).

I'm not hopeful, as I couldn't see any reference to the related interface pins in the microcode, but would like to be sure, as I'm not an electronics engineer nor have more than an AVO meter for diagnosis.

There's no great urgency about this, please reply as and when you have time.

Regards,
Tim

**Comment by keirf** [comment]
There aren't enough interface pins connected to the MCU I'm afraid. Most of those pins are grounds or NC in the Gotek schematic.

To get it working would either require hacking the Gotek PCB to connect to convenient MCU pins (eg. either directly, or to easier-to-solder pads such as the SPI Flash footprint). Or it would require a new Gotek schematic and PCB. And of course it would require firmware changes to recognise these new pins.

**Comment by jtgans** [comment]
I think we can solve this with an adapter board to solve the common cases for 1.44MB and 720KB images. 2.88MB images would be much harder and require a jumper from somewhere on the Gotek. I've started hacking something together in my PS2FF adapter board here: https://github.com/jtgans/PS2FF. Note that this is not currently working because the behavior on the media sense pins aren't clear, so don't expect to see these work correctly just yet.

**Reply by apaterak** [reply]
Hi! Any progress planned for this?

**Reply by jtgans** [reply]
It's open source. I need a timing diagram of the bus to figure out what is going on during the sense setup on POST to work out why the board isn't working. If you can supply that, that would be helpful, otherwise you're going to have to wait for me to get some time to fix my HP 16500. Sorry!
